### DISCUSSION TITLE
Custom configuration for 404 errors

### DISCUSSION THREAD

**Discussion by tobihans** [feathersjs/feathers#2567] [General]
Hi! I have a feathers application where I want to serve a SPA at the root. I also have some services I've created. So what I want to do is configure 404 errors to serve the index file if the route doesn't contain `/api`. But after many researches, I can't find a way to configure the `notFound` error handler.

**Comment by tobihans** [comment]
I got it. I just replaced it with a route matching `*` which sends the file.

**Comment by daffl** [comment]
The `notFound` handler just throws a 404 error. What you want to do is configure the `errorHandler` which is [documented here](https://docs.feathersjs.com/api/express.html#app-use-express-errorhandler-options). You can pass a filename or a complete custom formatter.

**Reply by tobihans** [reply]
Great, thanks 👍!
